File: scraping-service/models/queued_scrape_job.go
// ScrapeSmith/scraping-service/models/queued_scrape_job.go

package models

import "time"

type QueuedScrapeJob struct {
	OrderID      string    `bson:"orderId"`
	UserID       string    `bson:"userId"`
	URL          string    `bson:"url"`
	AnalysisType string    `bson:"analysisType"`
	CustomScript string    `bson:"customScript,omitempty"`
	CreatedAt    time.Time `bson:"createdAt"`
	Status       string    `bson:"status"`
	Attempts     int       `bson:"attempts"`
	LastTriedAt  time.Time `bson:"lastTriedAt,omitempty"`
}

File: job-service/src/models/QueuedScrapeJob.js
// ScrapeSmith/job-service/src/models/QueuedScrapeJob.js

import mongoose from 'mongoose';

const queuedScrapeJobSchema = new mongoose.Schema({
  orderId: { type: String, required: true, unique: true },
  userId: { type: String, required: true },
  url: { type: String, required: true },
  analysisType: { type: String, required: true },
  customScript: { type: String, default: null },
  createdAt: { type: Date, required: true, default: Date.now },
  status: {
    type: String,
    enum: ['pending', 'processing', 'done', 'failed'],
    default: 'pending'
  },
  attempts: { type: Number, default: 0 },
  lastTriedAt: { type: Date, default: null }
});

export const QueuedScrapeJob = mongoose.model('QueuedScrapeJob', queuedScrapeJobSchema);
